### DISCUSSION TITLE
ics files with date and time - can be used to add operationshours of waste disposal station

### DISCUSSION THREAD

**Discussion by bbilek** [mampfes/hacs_waste_collection_schedule#878] [Ideas]
First of all a big thank you !!! it works like a charm, despite some issues i had at the beginning to get it setup. 

Coming to the point: I would like to use this great piece of work, to also get open times of disposal stations made visible - by playing around with ics files (import / export) it looks like, that the times from an entry get removed - is there a chance to make them "visible" ? 

thank you !!

**Comment by 5ila5** [comment]
You are right, the ics source works fundamentally the same as other sources. The ICS file is processed only the start time will be processed and converted to a date object:
 https://github.com/mampfes/hacs_waste_collection_schedule/blob/f36e803544ee76cda25e123ee7ac49ea33d34852/custom_components/waste_collection_schedule/waste_collection_schedule/service/ICS.py#L48-L51
All events will be used to create a collection object containing all information about each event. 
https://github.com/mampfes/hacs_waste_collection_schedule/blob/f36e803544ee76cda25e123ee7ac49ea33d34852/custom_components/waste_collection_schedule/waste_collection_schedule/source/ics.py#L200-L205
Internally there is always a date representation used not a datetime, the end date is ignored completely. So implementing custom ics start and end times would mean a lot of changes. I do not think that this is currently any priority

**Comment by bbilek** [comment]
Thanks for getting back to me - i fully understand, that this is a bunch of work .... but despiteof this i would love the result ;-)
